packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramRenderer.tsx

```diff
@@ -151,6 +155,10 @@ export const DiagramRenderer = ({ diagram, selection, setSelection }: DiagramRen
     });
   };
 
+  const onEdgeUpdate = (oldEdge, newConnection) => {
+    reconnectEdge(oldEdge, newConnection);
```

Comment:
Be careful, a react flow Connection is not a react flow Edge.
You may need to update the `reconnectEdge` api so support Connection instead of Edge for the second argument